.github/workflows/deploy-nist.yml

```diff
@@ -0,0 +1,159 @@
+      - name: Set the environment based on the branch
+        id: define_environment
+        run: |
+          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
+            echo "env_name=staging" >> $GITHUB_OUTPUT
+          elif [ "${{ github.ref }}" = "refs/heads/production" ]; then
+            echo "env_name=production" >> $GITHUB_OUTPUT
+          fi
+      - name: Print the environment
+        run: echo "The environment is ${{ steps.define_environment.outputs.env_name }}"
+
+  build:
+    runs-on: ubuntu-latest
+    needs: define-environment
+    environment: ${{ needs.define-environment.outputs.env_name }}
+
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: 'nist-interface/'
+          sparse-checkout-cone-mode: true
+
+      - name: Read Node.js version from .nvmrc
+        id: nvm
+        run: echo "::set-output name=NODE_VERSION::$(cat nist-interface/.nvmrc)"
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ steps.nvm.outputs.NODE_VERSION }}
+
+      - name: Install Yarn
+        run: npm install -g yarn
+
+      - name: Build for production 🔧
+        working-directory: ./nist-interface
+        run: |
+          FEATURES_API_URL=$(grep '^FEATURES_API_URL=' ../.env | cut -d '=' -f 2)
+          echo REACT_APP_MAPBOX_TOKEN="${{ secrets.REACT_APP_MAPBOX_ACCESS_TOKEN }}" >> .env
+          echo REACT_APP_MAPBOX_STYLE_URL="${{ secrets.REACT_APP_MAPBOX_STYLE_URL }}" >> .env
+          echo REACT_APP_FEATURES_API_URL="${FEATURES_API_URL}" >> .env
+          echo "REACT_APP_PUBLIC_URL=$(echo "${{ vars.PUBLIC_URL_NIST }}" | sed -E 's|^https?://[^/]+||; s|/*$|/|')" >> .env
+          echo "Checking the REACT_APP_FEATURES_API_URL value in the .env file:"
+          grep 'REACT_APP_FEATURES_API_URL' .env
+          yarn
+          yarn build
+
+      - name: Upload build folder
+        uses: actions/upload-artifact@v4
+        with:
+          name: build_folder
+          path: ./nist-interface/build
+
+  deploy-staging:
+    needs:
+      - build
+      - define-environment
+    if: github.ref == 'refs/heads/main'
+    environment: ${{ needs.define-environment.outputs.env_name }}
+    
+    permissions:
+      id-token: write
+      contents: read
+
+    runs-on: ubuntu-latest
+    steps:
+      
+      - name: Checkout 🛎️
+        uses: actions/checkout@v4
+
+      - name: Download build folder
+        uses: actions/download-artifact@v4
+        with:
+          name: build_folder
+          path: ./nist-interface/build
+
+      - name: ConfigureAWS Credentials 
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.DEPLOYMENT_ROLE_ARN }}
+          role-session-name: ${{ github.repository_owner}}
+          aws-region: us-west-2
+      
+      - name: Upload to S3
+        run: |
+          export PUBLIC_URL=$(echo ${{ vars.PUBLIC_URL_NIST }} | sed -E 's|^https?://[^/]+/||')
```

Comment:
let's use the base_path from the github var directly instead of using sed to extract path from the url.